libs/native/src/main/java/org/elasticsearch/nativeaccess/CloseableByteBuffer.java

```diff
@@ -0,0 +1,18 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.nativeaccess;
+
+import java.nio.ByteBuffer;
+
+public interface CloseableByteBuffer extends AutoCloseable {
+    ByteBuffer buffer();
+
+    @Override
```

Comment:
Just a curiosity on my side: why override close here?

Reply:
The close method on AutoCloseable throws Exception. By overriding it here I reduce the checked exceptions (to none). Otherwise try-with-resources blocks that use CloseableByteBuffer would have to catch Exception, even though the two implementations don't actually throw anything.

Reply:
That's thoughtful, nice!